tasklist/els-schema/src/main/java/io/camunda/tasklist/es/ILMPolicyUpdateElasticSearch.java

```diff
@@ -0,0 +1,82 @@
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
+ */
+package io.camunda.tasklist.es;
+
+import io.camunda.tasklist.data.conditionals.ElasticSearchCondition;
+import io.camunda.tasklist.management.ILMPolicyUpdate;
+import io.camunda.tasklist.schema.manager.ElasticsearchSchemaManager;
+import java.util.Set;
+import java.util.regex.Pattern;
+import org.elasticsearch.client.indexlifecycle.GetLifecyclePolicyRequest;
+import org.elasticsearch.client.indexlifecycle.GetLifecyclePolicyResponse;
+import org.elasticsearch.common.settings.Settings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Component;
+
+@Component
+@Conditional(ElasticSearchCondition.class)
+public class ILMPolicyUpdateElasticSearch implements ILMPolicyUpdate {
+  private static final String TASKLIST_DELETE_ARCHIVED_INDICES = "tasklist_delete_archived_indices";
+  private static final String INDEX_LIFECYCLE_NAME = "index.lifecycle.name";
+  private static final String ARCHIVE_TEMPLATE_PATTERN_NAME_REGEX =
+      "^tasklist-.*-\\d+\\.\\d+\\.\\d+_\\d{4}-\\d{2}-\\d{2}$";
+  private static final Logger LOGGER = LoggerFactory.getLogger(ILMPolicyUpdateElasticSearch.class);
+  private static final String TASKLIST_PREFIX = "tasklist-*";
```

Comment:
`getIndexPrefix` cant be referred here because it is a not static property

As I wanna get the prefix + wildcard (*)
I have updated to 

`  private static final String TASKLIST_PREFIX_WILDCARD = TasklistElasticsearchProperties.DEFAULT_INDEX_PREFIX+"-*";`

Same will apply for the others.

Reply:
@nathansandi , you don't need to use a static property
you can autowire `TasklistElasticsearchProperties`, get the index prefix, and create the pattern at runtime

Reply:
I would prefer keep as the `static` property on the implementation, as this suffix is a constant, and this static property is exported from the `TasklistProperties`, 
as I am building a wildcard pattern with that. 
It is easier to maintain, and not necessary concatenate strings over there.

Reply:
@houssain-barouni  thanks for the clarification, 

Gonna work on the update.

Reply:
@houssain-barouni  updated over here https://github.com/camunda/zeebe/pull/17509/commits/33cbd32d38167802452e07e478374f83c962c2b2 

Understood the reason we need real time, as on SM we can change the suffix.
Thank you for note that.